src/log.c

```diff
@@ -206,24 +145,37 @@ static int writeEntry(Log *log, raft_entry_t *ety)
     size_t offset = FileSize(&log->file);
     size_t idxoffset = FileSize(&log->idxfile);
 
-    off += writeLength(buf + off, sizeof(buf) - off, '*', ENTRY_ELEM_COUNT);
-    off += writeString(buf + off, sizeof(buf) - off, ENTRY_STR);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->term);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->id);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->type);
-    off += writeLength(buf + off, sizeof(buf) - off, '$', ety->data_len);
+    off = generateEntryHeader(ety, buf, sizeof(buf));
 
     if (FileWrite(&log->file, buf, off) != (ssize_t) off ||
         FileWrite(&log->file, ety->data, ety->data_len) != ety->data_len ||
         FileWrite(&log->file, "\r\n", 2) != 2) {
         goto error;
     }
 
+    /* accumulating crc, so start with current crc */
+    long crc = crc16_ccitt(log->current_crc, buf, off);
+    crc = crc16_ccitt(crc, ety->data, ety->data_len);
+    crc = crc16_ccitt(crc, "\r\n", 2);
+
+    /* write crc as added element */
+    off = 0;
+    off += writeLong(buf + off, sizeof(buf) - off, crc);
+    if (FileWrite(&log->file, buf, off) != (ssize_t) off) {
+        goto error;
+    }
+
+    /* accumulating crc, so add the crc value we just wrote as well */
+    crc = crc16_ccitt(crc, buf, off);
+
     ssize_t ret = FileWrite(&log->idxfile, &offset, sizeof(offset));
     if (ret != sizeof(offset)) {
         goto error;
     }
 
+    /* everything succeeded, so can update accumulated crc */
+    log->current_crc = crc;
```

Comment:
Where do you keep `current_crc` - the crc of the whole log file?

Reply:
we don't, as discussed, I don't think it adds value, but still open to discussion.  It's very possible that "current_crc" we have is overkill and no need to really include the data of the CRC we write to the file, just a chain of the entries.  (so it wont correspond to the file, will just be in a sense a "block chain".)

Reply:
Yes, after a discussion with Yossi I agree with that.

Reply:
is that also about simplifying the crc and not recalculating it with the "crc" element.  i.e. just a chain of the data as mentioned?